docs/guides/src/main/server/proxy.adoc

```diff
@@ -0,0 +1,28 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/options.adoc" as opts>
+
+<@tmpl.guide
+title="Proxy setup"
+summary="Understand how to configure Keycloak behind a reverse proxy"
+includedOptions="proxy proxy.*">
+
+All proxy configuration options have a `proxy` prefix. You can use different proxy modes depending the proxy and TLS termination in your specific environment.
```

Comment:
Good as an overview, but as a guide, I would also expect more information such as:

* What headers should be forwarded by the reverse proxy. For instance, `X-Forward-*`. Useful to properly set-up configure ethe reverse proxy.
* Proxy usually means enabling load balancing. IMO, we should mention about session stickness. In fact, there is a flag that I think we should enable by default so that entries are created locally on each node and cookies are not appended with the route. 
* Perhaps explain why you should enable proxy in the first place?

Reply:
- added a paragraph about when to use it at the start
- added points about configuring the proxy itself, together with slightly modified note we have in current docs. (only tried to shorten it a bit)
- Not sure about mentioning sticky sessions here, could be too much for one guide. perhaps have another "tuning performance on a multi-node setup with sticky sessions" guide instead? wdyt?

Reply:
w.r.t. the flag, do you mean the `-Djboss.node.name=<nodename>` flag?

Reply:
> * Not sure about mentioning sticky sessions here, could be too much for one guide. perhaps have another "tuning performance on a multi-node setup with sticky sessions" guide instead? wdyt?

Nice to see how many steps/configuration we don't need here if compared with today :)

Yeah, perhaps we can have stick sessions covered in another guide, but my point is that configuring a proxy is also related to preparing Keycloak to properly work with sticksessions. For instance, https://www.keycloak.org/docs/latest/server_installation/#disable-adding-the-route.

Although that part is linked to "clustering" for me it is something related to proxy setup because it changes how cookies are created where the route is added by Keycloak with a reference to the node to where next requests should be redirected to. And most of the time, you want that disabled so that you just follow the LB strategies from your proxy.

In fact, we should probably change the default for `shouldAttachRoute` to `false`. And then we don't need to mention anything about that in any docs. Just mention that for HA you should consider enabling an LB strategy in the proxy.

@mposolda What do you think about making that option disabled by default?

Reply:
Created https://github.com/keycloak/keycloak/issues/9393.